cvat-sdk/cvat_sdk/pytorch/__init__.py

```diff
@@ -0,0 +1,280 @@
+import base64
+import collections
+import json
+import os
+import shutil
+import types
+import zipfile
+from concurrent.futures import ThreadPoolExecutor
+from pathlib import Path
+from typing import (
+    Callable,
+    Dict,
+    FrozenSet,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+)
+
+import appdirs
+import attrs
+import attrs.validators
+import PIL.Image
+import torchvision.datasets
+from typing_extensions import TypedDict
+
+import cvat_sdk.core
+import cvat_sdk.core.exceptions
+from cvat_sdk.api_client.model_utils import to_json
+from cvat_sdk.core.utils import atomic_writer
+from cvat_sdk.models import DataMetaRead, LabeledData, LabeledImage, LabeledShape, TaskRead
+
+_ModelType = TypeVar("_ModelType")
+
+_CACHE_DIR = Path(appdirs.user_cache_dir("CVAT_SDK", "CVAT.ai"))
+_NUM_DOWNLOAD_THREADS = 4
+
+
+class UnsupportedDatasetError(cvat_sdk.core.exceptions.CvatSdkException):
+    pass
+
+
+@attrs.frozen
+class FrameAnnotations:
+    tags: List[LabeledImage] = attrs.Factory(list)
+    shapes: List[LabeledShape] = attrs.Factory(list)
+
+
+@attrs.frozen
+class Target:
+    annotations: FrameAnnotations
+    label_id_to_index: Mapping[int, int]
+
+
+class TaskDataset(torchvision.datasets.VisionDataset):
```

Comment:
Where can I find ProjectDataset?

Reply:
There is none at the moment; however, one could be built easily using [`ConcatDataset`](https://pytorch.org/docs/stable/data.html#torch.utils.data.ConcatDataset).

Frankly, I'm not sure if it would even make sense to create a `Dataset` out of the entire project - such a dataset would contain all of the subsets, so it wouldn't be suitable for either training or testing.